Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,180 @@ Finally, this part describes the OAuth security applied to the API. This spec is
 - Endpoint token URL
 - URL to endpoint authorization ( If flow is based on "`authorizationCode`").	
 
+
+## 12. Subscription, Notification & Event
+
+In order to provided event-based interaction, CAMARA API could provided capabilities for subscription & notification management.
+A subscription allows to an API consumer to request event notification reception at an given url (callback address) for a specific context.
+A notification is the publication at the listener address about an occurred event.
+Event type managed are explicitly defined in CAMAPA API OAS.
+
+### 12.1 Subscription
+
+We distinguish 2 types of subscription:
+- Instance-based subscription (indirect creation)
+- Resource-based subscription (direct creation)
+
+**Instance-based subscription**
+
+An instance-based subscription is a subscription indirectly created, additionnally, to another resource creation. For example for a Payment request (in Carrier Billing API), in the `POST/payments`, the API consumer could request to get event notification about **this** Payment request processing update. The subscription is not an entity and its lifecycle is linked to the managed entity (the Payment resource in this case). The subscription terminates with the managed entity.
+
+Providing this capability is optionnal for any CAMARA APIs depending on UC requirements
+
+If this capability is present in CAMARA API, following attributes **must** be used in the POST request for the managed entity:
+
+| attribute name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  | -----  | 
+| notificationUrl | string |callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | authentification token for callback API | optional |
+
+
+**Resource-based subscription**
+
+A resource-based subscription is is subscription managed as a resource. A dedicated endpoint is provided to request subscription creation.  As this subscription is managed API resource, it is identified and operations to search, retrieve and delete them must be provided.
+
+Note: It is perfectly valid for a CAMARA API to have several subscriptions endpoints when distinct events are managed (for example in Device status API we manage distinct subscription for roaming status and reachability status event)
```

Comment:
This is a good point to be discussed. Indeed we have 2 alternatives in each API managing events:

- providing a specific endpoint for subscription : `POST /subscriptions/roaming` (and in future for this same device status API add `POST /subscriptions/connectivityStatus`)
- providing an generic endpoint requiring in the body the eventType ( `POST /subscriptions/`)

I have no strong feeling about one or the other but I think we should not provide both. We should have one pattern and use if for all API.

To be noted that for GET & DELETE operation the proposal is to use generic path.

Reply:
I also think we should define both models. At the end the objective here in commonalities is to define a pattern that we can use within every API may need.

Which pattern to use/define in every API (one, other or both), will depend on the Use Case/s within the business logic  associated. That is for internal discussion within each WG that may apply

So it makes full sense to define both models:
* Instance-based subscription (indirect creation)
* Resource-based subscription (direct creation)

Reply:
Took into account in today update.

Reply:
I'm still of the opinion that the format /subscriptions/<subscriptionName> is in conflict with this guideline, which follows the general REST-API recommendations: It means  `/<entity>/{<entityId>}`
In this case it would be /subscriptions/<subscription-id

I think if we only want to specify one way of endpoint definition, that it should be this way.